source/frontend/src/features/admin/financial-codes/list/FinancialCodeListView.tsx

```diff
@@ -0,0 +1,131 @@
+import { Roles } from 'constants/roles';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import orderBy from 'lodash/orderBy';
+import { Api_FinancialCode } from 'models/api/FinancialCode';
+import moment from 'moment';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaPlus } from 'react-icons/fa';
+import { useHistory } from 'react-router';
+import styled from 'styled-components';
+
+import { useFinancialCodeRepository } from '../hooks/useFinancialCodeRepository';
+import FinancialCodeFilter, {
+  defaultFinancialCodeFilter,
+  IFinancialCodeFilter,
+} from './FinancialCodeFilter/FinancialCodeFilter';
+import { FinancialCodeResults } from './FinancialCodeResults/FinancialCodeResults';
+import * as Styled from './styles';
+
+/**
+ * Page that displays acquisition files information.
+ */
+export const FinancialCodeListView: React.FC = () => {
+  const history = useHistory();
+  const { hasRole } = useKeycloakWrapper();
+  const {
+    getFinancialCodes: {
+      execute: getFinancialCodes,
+      response: financialCodeResults,
+      loading: financialCodesLoading,
+    },
+  } = useFinancialCodeRepository();
+
+  const fetchData = useCallback(async () => {
+    await getFinancialCodes();
+  }, [getFinancialCodes]);
+
+  useEffect(() => {
+    if (financialCodeResults === undefined) {
+      fetchData();
+    }
+  }, [fetchData, financialCodeResults]);
+
+  // Sorting and filtering for this list view is performed client-side
+  const [sort, setSort] = React.useState<TableSort<Api_FinancialCode>>({});
+  const [filter, setFilter] = React.useState<IFinancialCodeFilter>(defaultFinancialCodeFilter);
+
+  const sortedFilteredFinancialCodes = useMemo(() => {
+    if (financialCodeResults && financialCodeResults?.length > 0) {
+      let records = [...financialCodeResults];
+      if (filter) {
+        records = records.filter(record => {
+          // Apply all the filters
+          if (filter.financialCodeType && record.type !== filter.financialCodeType) {
+            return false;
+          }
+          if (
+            filter.codeValueOrDescription &&
+            !record.code?.toLowerCase().includes(filter.codeValueOrDescription.toLowerCase()) &&
+            !record.description?.toLowerCase().includes(filter.codeValueOrDescription.toLowerCase())
+          ) {
+            return false;
+          }
+          if (filter.showExpiredCodes === false && isExpiredCode(record)) {
+            return false;
+          }
+          // Finally return true because record has satisfied all conditions
+          return true;
+        });
+      }
+
+      if (sort) {
+        const sortFields = Object.keys(sort);
+        if (sortFields?.length > 0) {
+          const sortBy = sortFields[0] as keyof Api_FinancialCode;
+          const sortDirection = sort[sortBy];
+          return orderBy(records, sortBy, sortDirection);
```

Comment:
This isn't working correctly for the code value column as it is sorting those values as if they were strings not numbers: 
![image](https://user-images.githubusercontent.com/41091511/212769533-2a1e1b94-5953-44c0-bad7-5be0f8cc7153.png)

Reply:
This the result of the code property on the JSON model being a string. If you look at the DB there are some tables that had number codes and others had string codes. One had to be picked up and I believe Doug's latest DB change (in review) will change all codes to be string

Reply:
If they want to sort as numbers (never specified in AC) we would need a different approach (most likely a new list view for codes that are numeric, not sure)